addons/website_sale/static/src/js/website_sale.js

```diff
@@ -312,8 +312,13 @@ odoo.define('website_sale.website_sale', function (require) {
             for (var k in variant_ids) {
                 if (_.isEmpty(_.difference(variant_ids[k][1], values))) {
                     $.when(base.ready()).then(function() {
+                        if ($price_confirmation.attr("data-product-id") == variant_ids[k][0]) {
```

Comment:
Yes, it could be true, when:
- you are in advanced pricing based on formulas (in sales config) AND
- when an element of a pricelist applies only on one product variant